lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/TestResubmitTask.cs

```diff
@@ -0,0 +1,227 @@
+            }
+
+            public void OnNext(IFailedTask value)
+            {
+                var failedExeption = ByteUtilities.ByteArraysToString(value.Data.Value);
+                Logger.Log(Level.Error, "In IFailedTask: " + failedExeption);
+                Assert.Equal(TaskId1, value.Id);
+                Assert.Contains(TaskKilledByDriver, failedExeption);
+                value.GetActiveContext().Value.SubmitTask(GetTaskConfigurationForCloseTask(TaskId2));
+            }
+
+            public void OnNext(IRunningTask value)
+            {
+                Logger.Log(Level.Info, "Task running: " + value.Id);
+                if (value.Id.Equals(TaskId1))
+                {
+                    value.Dispose(Encoding.UTF8.GetBytes(KillTaskCommandFromDriver));
+                }
+                else if (value.Id.Equals(TaskId2))
+                {
+                    value.Send(Encoding.UTF8.GetBytes(CompleteTaskCommandFromDriver));
+                }
+                else
+                {
+                    throw new Exception("It should not be reached.");
+                }
+            }
+
+            public void OnCompleted()
+            {
+                throw new NotImplementedException();
+            }
+
+            public void OnError(Exception error)
+            {
+                throw new NotImplementedException();
+            }
+
+            private IConfiguration GetTaskConfigurationForCloseTask(string taskId)
+            {
+                return TaskConfiguration.ConfigurationModule
+                    .Set(TaskConfiguration.Identifier, taskId)
+                    .Set(TaskConfiguration.Task, GenericType<ResubmitTask>.Class)
+                    .Set(TaskConfiguration.OnClose, GenericType<ResubmitTask>.Class)
+                    .Set(TaskConfiguration.OnMessage, GenericType<ResubmitTask>.Class)
+                    .Build();
+            }
+        }
+
+        private sealed class ResubmitTask : ITask, IDriverMessageHandler, IObserver<ICloseEvent>
+        {
+            private readonly CountdownEvent _suspendSignal = new CountdownEvent(1);
+
+            [Inject]
+            private ResubmitTask()
+            {
+            }
+
+            public byte[] Call(byte[] memento)
+            {
+                Logger.Log(Level.Info, "Hello in ResubmitTask");
+                _suspendSignal.Wait();
+                return null;
+            }
+
+            public void Dispose()
+            {
+                Logger.Log(Level.Info, "Task is disposed.");
+            }
+
+            public void OnNext(ICloseEvent value)
+            {
+                if (value.Value != null && value.Value.Value != null)
+                {
+                    Logger.Log(Level.Info,
+                        "Closed event received in task:" + Encoding.UTF8.GetString(value.Value.Value));
+                    if (Encoding.UTF8.GetString(value.Value.Value).Equals(KillTaskCommandFromDriver))
```

Comment:
@jwang98052 Throwing an exception does not work in the default case, because that would be the expected behavior of this class, even though it doesn't get the right message. The right thing to do here is to log a failure, signal the wait event, and let the task complete (since that is the opposite of what we expect from the failure task).